src/main/java/org/embulk/parser/avro/AvroParserPlugin.java

```diff
@@ -128,26 +133,14 @@ public void run(TaskSource taskSource, Schema schema,
         PluginTask task = taskSource.loadTask(PluginTask.class);
         List<Column> columns = schema.getColumns();
         final TimestampParser[] timestampParsers = Timestamps.newTimestampColumnParsers(task, task.getColumns());
-        File avsc = task.getAvsc().getFile();
-        final org.apache.avro.Schema avroSchema;
-        try {
-            avroSchema = new org.apache.avro.Schema.Parser().parse(avsc);
-        } catch (IOException e) {
-            throw new ConfigException("avsc file is not found");
-        }
 
         try (FileInputInputStream is = new FileInputInputStream(input); final PageBuilder pageBuilder = new PageBuilder(Exec.getBufferAllocator(), schema, output)) {
-            ColumnGetterFactory factory = new ColumnGetterFactory(avroSchema, pageBuilder, timestampParsers);
-            ImmutableMap.Builder<String, BaseColumnGetter> columnGettersBuilder = ImmutableMap.builder();
-            for (Column column : columns) {
-                BaseColumnGetter columnGetter = factory.newColumnGetter(column);
-                columnGettersBuilder.put(column.getName(), columnGetter);
-            }
-            ImmutableMap<String, BaseColumnGetter> columnGetters = columnGettersBuilder.build();
-            DatumReader<GenericRecord> reader = new GenericDatumReader<>(avroSchema);
+            ColumnGetterFactory factory = new ColumnGetterFactory(pageBuilder, timestampParsers);
+            DatumReader<GenericRecord> reader = new GenericDatumReader<>();
```

Comment:
A reader can read avro file by different schema from one of itself.
For example, avro file doesn't have `new_column`, but reader can read `new_column` with default value by new schema.
Unless embulk plugin passes avroSchema object to `GenericDatumReader`, plugin user cannot use this feature.
And so we need more hacks.

Reply:
I see. I've supported such cases in https://github.com/joker1007/embulk-parser-avro/pull/2/commits/d792c3f7c8463c2188bea2ec65d2a0fda9d37742. How about it?